sklearn/linear_model/tests/test_sgd.py

```diff
@@ -57,19 +64,39 @@ def predict_proba(self, X):
         return super().predict_proba(X)
 
 
-class SparseSGDRegressor(SGDRegressor):
-
+class _SparseSGDRegressor(linear_model.SGDRegressor):
```

Comment:
I still find these strange but better than before ...

Reply:
basically I need 4 factories. I could not come up with a more elegant solution.